src/Client/StatusWatcherClient/StatusWatcherAnalyzer.php

```diff
@@ -118,6 +118,7 @@ private function analyzeCurrentStatuses(AnonymousMessage $message): void
         $contacts = new CurrentContacts($message);
         foreach ($contacts->getUsers() as $user) {
             $this->performStatusReaction($user->getUserId(), $user->getStatus());
+            $this->notifier->onUserNameChange($user->getUserId(), $user->getUsername());
```

Comment:
Watcher should generate updates only if usernames actually changed

Reply:
Fixed